app/controllers/my_module_repository_snapshots_controller.rb

```diff
@@ -30,22 +30,38 @@ def index_dt
   end
 
   def create
-    service = Repositories::MyModuleAssigningSnapshotService.call(repository: @repository,
-                                                                  my_module: @my_module,
-                                                                  user: current_user)
-
-    if service.succeed?
-      @repository_snapshots = @my_module.repository_snapshots.where(original_repository: @repository)
-      render json: { html: render_to_string(partial: 'my_modules/repositories/full_view_versions_sidebar') }
-    else
-      render json: service.errors, status: :unprocessable_entity
-    end
+    repository_snapshot = @repository.dup.becomes(RepositorySnapshot)
+    repository_snapshot.assign_attributes(type: RepositorySnapshot.name,
+                                          original_repository: @repository,
+                                          my_module: @my_module,
+                                          created_by: current_user)
+    repository_snapshot.provisioning!
+    repository_snapshot.reload
+
+    RepositorySnapshotProvisioningJob.perform_later(repository_snapshot)
+
+    render json: {
+      html: render_to_string(partial: 'my_modules/repositories/full_view_version',
+                             locals: { repository_snapshot: repository_snapshot })
+    }
+  end
+
+  def status
+    render json: {
+      status: @repository_snapshot.status
+    }
+  end
+
+  def show
+    render json: {
+      html: render_to_string(partial: 'my_modules/repositories/full_view_version',
+                             locals: { repository_snapshot: @repository_snapshot })
+    }
   end
 
   def destroy
     @repository_snapshot.destroy!
-    @repository_snapshots = @my_module.repository_snapshots.where(original_repository: @repository)
-    render json: { html: render_to_string(partial: 'my_modules/repositories/full_view_versions_sidebar') }
+    render json: {}
```

Comment:
Would at least ID of deleter record come handy at some point?

Reply:
Don't think so